packages/react-components/react-dialog/Spec.md

```diff
@@ -1,63 +1,459 @@
+2. **(2 & 3)** Focus is automatically set to the first focusable element inside the dialog
+3. **EnterKey** Confirms or cancels the alert message and dialog is dismissed.
+4. **TabKey** Moves focus to next focusable element inside the dialog. When focus is on the last focusable element in the dialog, moves focus to the next focusable action in the browser window.
+5. **Shift+Tab** Moves focus to previous focusable element inside the dialog. When focus is on the first focusable element in the dialog, moves focus to the last focusable action within the browser window.
 
 ## Accessibility
 
-Base accessibility information is included in the design document. After the spec is filled and review, outcomes from it need to be communicated to design and incorporated in the design document.
-
-- Decide whether to use **native element** or folow **ARIA** and provide reasons
-- Identify the **[ARIA](https://www.w3.org/TR/wai-aria-practices-1.2/) pattern** and, if the component is listed there, follow its specification as possible.
-- Identify accessibility **variants**, the `role` ([ARIA roles](https://www.w3.org/TR/wai-aria-1.1/#role_definitions)) of the component, its `slots` and `aria-*` props.
-- Describe the **keyboard navigation**: Tab Oder and Arrow Key Navigation. Describe any other keyboard **shortcuts** used
-- Specify texts for **state change announcements** - [ARIA live regions
-  ](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Live_Regions) (number of available items in dropdown, error messages, confirmations, ...)
-- Identify UI parts that appear on **hover or focus** and specify keyboard and screen reader interaction with them
-- List cases when **focus** needs to be **trapped** in sections of the UI (for dialogs and popups or for hierarchical navigation)
-- List cases when **focus** needs to be **moved programatically** (if parts of the UI are appearing/disappearing or other cases)
+> ⚠️ _Note: All other accessibility information, not covered in this section, is provided throughout the spec._
+
+The dialog component follows the [Dialog WAI-Aria design pattern](https://www.w3.org/WAI/ARIA/apg/patterns/dialogmodal/).
+
+### Aria roles and states
+
+#### Modal
+
+- Trigger button
+  - [`aria-haspopup="dialog"`](https://w3c.github.io/aria/#aria-haspopup)
+    > ⚠️ This is deprecated, the proper attribute should be [`aria-expanded=true`](https://w3c.github.io/aria/#aria-expanded)
+- Dialog
+  - [`role="dialog"`](https://w3c.github.io/aria/#dialog)
+  - [`aria-modal=true`](https://w3c.github.io/aria/#aria-modal)
+  - [`aria-labelledby={dialog-title-idref}`](https://w3c.github.io/aria/#aria-labelledby)
+  - [`aria-label="some label"`](https://w3c.github.io/aria/#aria-label)
+
+#### Non-modal
+
+- Trigger button
+  - [`aria-haspopup="dialog"`](https://w3c.github.io/aria/#aria-haspopup)
+    > ⚠️ This is deprecated, the proper attribute should be [`aria-expanded=true`](https://w3c.github.io/aria/#aria-expanded)
```

Comment:
can you please add a link to the deprecation? as far as I understand, aria-haspopup is still required - https://www.w3.org/TR/wai-aria/#aria-haspopup - even though the support is insufficient - https://a11ysupport.io/tech/aria/aria-haspopup_attribute

Reply:
Isn't that the link https://w3c.github.io/aria/#aria-haspopup? 
![image](https://user-images.githubusercontent.com/5483269/173061741-0901e799-9db2-47d5-a228-303223bb69f9.png)

Reply:
it is being depreacted as a global property. it will be allowed for relevant roles, I believe button is a relevant role for it.